### ISSUE TITLE
Load tests

### ISSUE THREAD

**Issue by tam7t** [GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp#32]
### TL;DR
We should load test (and document bottlenecks) to mount events.

Currently, a single mount event will exec our binary which
- **kube-apiserver** `get` `serviceaccounts` (?)
- **kube-apiserver** `create` `serviceaccounts/token` (?)
- **GCP IAM** `GenerateAccessToken` ([60k QPM Quota](https://cloud.google.com/iam/quotas))
- **GCP SM** `AccessSecretVersion` ([90k QPM Quota](https://cloud.google.com/secret-manager/quotas#request-rate-quotas))

### Design

**Proposal**

Having target performance goals could influence design. A long-lived plugin (rather than an exec'd one) could have an opportunity to:
1) re-use network/http connections across mount events
2) cache SA information + tokens

If `kube-apiserver` can handle 60K qpm of requests - and the target performance goals do not exceed that number then no architecture changes would be needed.

**Alternatives considered**

**Resources**

**Additional information**

https://github.com/kubernetes-sigs/secrets-store-csi-driver/issues/257
